roxie/roxiemem/roxiemem.cpp

```diff
@@ -224,10 +224,13 @@ typedef MapBetween<unsigned, unsigned, memsize_t, memsize_t> MapActivityToMemsiz
 
 static CriticalSection heapBitCrit;
 
-static void initializeHeap(bool allowHugePages, bool allowTransparentHugePages, bool retainMemory, memsize_t pages, memsize_t largeBlockGranularity, ILargeMemCallback * largeBlockCallback)
+static void initializeHeap(bool allowHugePages, bool allowTransparentHugePages, bool retainMemory, bool lockMemory, memsize_t pages, memsize_t largeBlockGranularity, ILargeMemCallback * largeBlockCallback)
 {
     if (heapBase) return;
 
+    if (lockMemory && !retainMemory)
+        retainMemory = true;
```

Comment:
Perhaps a warning that lockMemory is being ignored because retainMemory was not set?

Reply:
We thought setting lockMemory should not also require setting retainMemory.
So there are 3 settings, none, retainMemory set by itself and lockMemory set (which implies retainMemory)

Reply:
Seems fair